### ISSUE TITLE
Add Nix

### ISSUE THREAD

**Issue by Ericson2314** [msys2/MINGW-packages#22459]
### Package name

Nix

### Brief description of package

Nix, the purely functional package manager.

Nix has a Windows port in progress, which I am involved with. It is not yet feature-complete, but it does build. We understand if packaging software that isn't yet useful is putting the cart before the horse, but we would love to better do native builds of Nix while we finish the port, and also learn a bit more about MSYS / toolchain setups in the process.

Once we're far enough to be attempting to build Nixpkgs on Windows natively with Cygwin/MSYS, we'll also need to learn a lot from you all in general. (See https://github.com/NixOS/nixpkgs/pull/354137 for the first steps on that.) So consider this a hybrid packaging requestion / general knowledge sharing and collaboration request.

(And of course, if someday you all want to ditch `pacman` for `nix`, we won't complain :))

### URL for package's homepage

https://github.com/nixos/nix

### Provide a basic test case to validate the package's functionality.

We have unit tests for the libraries that run, and some in-progress work to run the shell-script-based integration tests.

### MINGW environments where you need the package

- [X] MINGW64
- [X] MINGW32
- [X] UCRT64
- [X] CLANG64
- [X] CLANGARM64

### Are you willing to submit a PR?

I think one of us porting Nix to Windows could do it but we would need hand-holding. Ideally one of you could drive the PR, and we would be highly available to answer questions.

**Comment by lazka** [comment]
Would nix target the Windows APIs via C++, or is this a cygwin port?

**Comment by Ericson2314** [comment]
@lazka No Cygwin! :tada: We're cross building from Linux with MinGW now, and a previous version of the port (https://github.com/nix-windows/nix/commit/windows) in fact got things building with MSVC.

The newish C++ `filesystem` library helps a lot, here :)

**Comment by puffnfresh** [comment]
I'll take this on, since I've got an [automated build of Nix using mingw64](https://github.com/nix-windows/nix/commit/f493fe78d4fbd54f2f12eb6c87e6b0fd63877e66#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR61-R107). There are two dependencies missing:

* [ ] toml11 #22409
* [ ] libgit2, with a patch which is only in HEAD, not in a release yet #6875 (I think we should just wait)

**Comment by Kreijstal** [comment]
> @lazka No Cygwin! 🎉 We're cross building from Linux with MinGW now, and a previous version of the port ([nix-windows/nix@windows](https://github.com/nix-windows/nix/commit/windows)) in fact got things building with MSVC.
> 
> The newish C++ `filesystem` library helps a lot, here :)

there are many programs still depending on autotools.. it would've be certainly easier if it was cygwin based. (Would support more from source built software as it is right now) But maybe there is less dependency on it than before.. We will see.

**Comment by Kreijstal** [comment]
> > @lazka No Cygwin! 🎉 We're cross building from Linux with MinGW now, and a previous version of the port ([nix-windows/nix@windows](https://github.com/nix-windows/nix/commit/windows)) in fact got things building with MSVC.
> > The newish C++ `filesystem` library helps a lot, here :)
> 
> there are many programs still depending on autotools.. it would've be certainly easier if it was cygwin based. (Would support more from source built software as it is right now) But maybe there is less dependency on it than before.. We will see.

Pd: maybe with nix, I can finally get guix :)

**Comment by lazka** [comment]
We've updated libgit2 not in #20396. In theory we could backport things if needed there.

**Comment by Ericson2314** [comment]
@lazka Yeah if you would be open to a patch, I think it is a pretty non-invasive change. https://github.com/libgit2/libgit2/compare/v1.8.4..main/#diff-069ad7b34be69df5ba08a066f481b6d5b5f473894137065f33eebfc2a7722442 shows it is not yet landed in that patch release. I'll see what I can do if @puffnfresh doesn't beat me to it :).

**Comment by lazka** [comment]
https://github.com/msys2/MINGW-packages/pull/22492

**Comment by Ericson2314** [comment]
Wonderful! Really appreciate that @lazka :)

**Comment by inschrift-spruch-raum** [comment]
May I take the liberty of asking how the project is currently progressing
